packages/fluentui/docs/src/examples/components/Checkbox/States/CheckboxExampleIndeterminate.shorthand.tsx

```diff
@@ -0,0 +1,42 @@
+import { Checkbox, CheckboxProps } from '@fluentui/react-northstar';
+import * as React from 'react';
+
+const CheckboxExampleChecked = () => {
+  const [lettuceChecked, setLettuceCheck] = React.useState(false);
+  const [tomatoChecked, setTomatoCheck] = React.useState(false);
+  let checked: CheckboxProps['checked'] = lettuceChecked && tomatoChecked;
+  if (!checked) {
+    checked = !lettuceChecked && !tomatoChecked ? false : 'mixed';
+  }
```

Comment:
I played with this example, I suggest to add a third checkbox to a group and use `useReducer()` instead:

```
const [state, dispatch] = ...
/* state: { lettuce, tomato, ?? }
```

To quickly compute state of the main checkbox you can use magic ternary 🌜 

```
const checked = Object.keys(state).every(Boolean) || (Object.keys(state).some(Boolean) ? 'mixed' : false)
```